New term entry for Java Math.sinh() (#1121)
Codecademy/docs: content/java/concepts/math-methods/terms/sinh/sinh.md

---
Title: '.sinh()'

> Description: 'Returns the hyperbolic sine of a double-type value. The hyperbolic sine of x is defined to be (e^x - e^-x)/2 where e is Euler's number.'

Reviewer: ```suggestion
Description: 'Returns the hyperbolic sine of a double-type value.'
```

Subjects:
  - 'Computer Science'
Tags:
  - 'Arithmetic'
  - 'Functions'
  - 'Methods'
  - 'Numbers'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

The **`Math.sinh()`** method returns the hyperbolic sine of a double-type value.

## Syntax

```pseudo
Math.sinh(double num)
```

> The return type of the `.sinh()` method is a double.

Reviewer 2: Types in Java should be surrounded by backticks.

Some special cases for the `.sinh()` method include the following:

- If the argument is `NaN`, then the result is `NaN`.
- If the argument is infinite, then the result is an infinity with the same sign as the argument.
- If the argument is zero, then the result is a zero with the same sign as the argument.

> The computed result will be within 2.5 ulps of the exact result.

Reviewer 2: What does "ulps" mean?

Author: Added an explanation for ULP.

---
Title: '.sinh()'
Description: 'Returns the hyperbolic sine of a double-type value.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Arithmetic'
  - 'Functions'
  - 'Methods'
  - 'Numbers'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> The **`Math.sinh()`** method returns the hyperbolic sine of a double-type value.

Reviewer 3: ```suggestion
The **`Math.sinh()`** method returns the hyperbolic sine of a `double`-type value.
```

## Syntax

```pseudo
Math.sinh(double num)
```

The return type of the `.sinh()` method is a `double`.

Some special cases for the `.sinh()` method include the following:

> - If the argument is `NaN`, then the result is `NaN`.
> - If the argument is infinite, then the result is an infinity with the same sign as the argument.
> - If the argument is zero, then the result is a zero with the same sign as the argument.

Reviewer 3: ```suggestion
- If `num` is `NaN`, then the result is `NaN`.
- If `num` is infinite, then infinity is returned with the same sign as `num`.
- If `num` is zero, then zero is returned with the same sign as `num`.
```

> The computed result will be within 2.5 ulps of the exact result.
> 
> An ulp stands for unit of least precision. It calculates the distance between the given `double` or `float` value and the
> `double` or `float` value next larger in magnitude.

Reviewer 3: ```suggestion
The computed result will be within 2.5 units of least precision (ulps) of the exact result.
```

The latter part of the text is more suitable for another entry (perhaps one about the `Math.ulp()` Java method).

Author: Added the latter part as the description was asked by @HishamT.
